### ISSUE TITLE
Docs redesign - new navigation structure + theming

### ISSUE THREAD

**Issue by Holben888** [slinkity/slinkity#103]
## New designs

Sidebar changes:
- **a nicer experience for mobile and tablet users** - a collapsable menu you can summon from anywhere on the page (instead of scrolling back to the top)
- **new section headings** to better surface our pages
- **a nicer a e s t h e t i c** in keeping with our retro theme 😎

Primary nav changes:
- a new purple background to match our new theme

Documentation content changes:
- a dull purple background to match our new theme

### Mobile and tablet breakpoints

Note: These should match with the `breakpoint(mobile)` and `breakpoint(tablet)` values under our SCSS functions
- [Link to Figma mobile mockup](https://www.figma.com/file/6sRiQ46r51pJKaVr11fB6I/Slinkity-docs?node-id=129%3A209)
- [Link to Figma tablet mockup](https://www.figma.com/file/6sRiQ46r51pJKaVr11fB6I/Slinkity-docs?node-id=129%3A70)

<img width="543" alt="image" src="https://user-images.githubusercontent.com/31811199/147773480-8d5948b7-511b-405b-bca2-012cc7fbea81.png">

### Desktop and above

- [Link to Figma desktop mockup](https://www.figma.com/file/6sRiQ46r51pJKaVr11fB6I/Slinkity-docs?node-id=119%3A2)

<img width="653" alt="image" src="https://user-images.githubusercontent.com/31811199/147773563-ce5080e4-e560-4827-80d3-1753b7cdc456.png">

## Code changes

- we'll likely need a new mixin to represent those nav section headings. I say "mixin" since it'll be a combo of `font-family`, `text-transform` to all caps, and wider `letter-spacing`. I'm open to any naming scheme here; maybe `subheading` or `decorative-heading`?
- we currently assume all visitors have the Monaco font on their device, but this probably isn't sustainable! We should probably find this font face and add it to our `assets/fonts` directory.
- the `docs.11tydata.js` will need a new data structure to support section headings. The section headings shown in the mockup _are_ intended for production (it's not dummy data!), so structure our data to support that. i18n is out of scope.

**Comment by Holben888** [comment]
To be completd after #102

**Comment by brycewray** [comment]
@Holben888 Re your second bullet under "Code changes," you could try something similar to [what Tailwind uses for monospaced type](https://tailwindcss.com/docs/font-family), unless you require that code will look exactly the same regardless of OS or device (*e.g.*, so all zeroes will look the same):

```css
font-family: ui-monospace, SFMono-Regular, Menlo, Monaco, Consolas, "Liberation Mono", "Courier New", monospace;
```

**Comment by Holben888** [comment]
@brycewray Ah that's fair! But we _are_ using Monaco as a heading font as well, so not just code snippets. I'd prefer to get the same look across device now that it's a decorative font.

**Comment by nachtfunke** [comment]
I think after merging #167, everything described in this issue is done! I meant to reference this issue in my PR but I referenced the wrong issue.

**Comment by Holben888** [comment]
No prob! I agree 😁
